### ISSUE TITLE
Set up SMTP on web server

### ISSUE THREAD

**Issue by thethomp** [thethomp/lammergeyer#21]
The setup of this should be automated and the script committed to the server branch. Should also look up basic way to send and receive email using this, possibly include this as a test script to verify the setup.

**Comment by thethomp** [comment]
Might be able to just use the Django email functions to handle email overall, but it will still require a base SMTP install.

**Comment by jmt4** [comment]
I've been thinking about this one a little bit and there seems to be at least two ways we can do this. Each with pros and cons.
1. As was mentioned, we send an email with the reminder and await the response (in the form of a reply?).
   - Con. We actually have to go through the reply process for an email. Admittedly this is not a big deal since the response could be empty.
   - Con. Presumably, we would put the reminder pk in the email heading so when the person replies, we could just parse that line and figure out which reminder to deactivate. This strategy opens the door for the email heading being modified and the reminder not being deactivated. Granted this would only affect the person with the reminder I believe. Maybe there could be a hidden field, I don't know enough about this yet.
   - Pro. We would have the email address and the reminder pk.
2. We send the reminder with a deactivation link. Similar to the confirmation email problem.
   - Con. Clicking on the link would open a new tab which I think would be a little annoying. Maybe you can make it so the tab auto closes I don't know.
   - Pro. Clicking on a link is easier than making a reply albeit not by much on our end and for the user.

Just some thoughts. Let me know what you think

**Comment by jmt4** [comment]
I guess emails can include html?? If this is the case, then we could include identification information in some sort of hidden input assuming we still have access to the original message in a reply.

**Comment by thethomp** [comment]
Yep, emails can definitely include HTML (this one does!) Good call with
that, can probably embed something in the html header or something, though
probably need to be able to handle the case of plain text email as well. I
like the idea of including a link to press as opposed to replying (which is
what I was considering), I think that's the better way to go personally, if
we can figure out how to do it.

**Comment by jmt4** [comment]
There is a high chance that we can re-use or modify some of django's [token generator](https://github.com/django/django/blob/master/django/contrib/auth/tokens.py) class to make confirmation codes. I haven't looked at this code in too much detail but the tokens are timestampable and even saltable which might actually be what we want.

We could also make a "behavioral mixin" that our models inherit from. Something simple
with fields for is_active and conf_code. And then of course the url .../reminders/deactivate/(conf_code) and view to handle it. This is doable I think.

If we want to log the user in when they click a deactivation link, I think the problem becomes quite a bit more complicated. I think there might be quite a few security issues to consider. Anyhow, we can talk about it in our next discussion.
